app/screens/map-screen/map-screen.tsx

```diff
@@ -70,38 +99,84 @@ export const MapScreen: React.FC<Props> = ({ navigation }) => {
+        const countryCodesToCoords: {
+          data: Record<CountryCode, { lat: number; lng: number }>
+        } = JSON.parse(JSON.stringify(countryCodes))
+        const countryCoords: { lat: number; lng: number } =
+          countryCodesToCoords.data[countryCode]
+        if (countryCoords) {
+          const region: Region = {
+            latitude: countryCoords.lat,
+            longitude: countryCoords.lng,
+            latitudeDelta: LATITUDE_DELTA,
+            longitudeDelta: LONGITUDE_DELTA,
+          }
+          setUserLocation(region)
+        }
+      }
+      setIsLoadingLocation(false)
+    }
+  }, [wasLocationDenied, countryCode, loading, setIsLoadingLocation, setUserLocation])
+
+  const getUserRegion = (callback: (region?: Region) => void) => {
+    try {
+      Geolocation.getCurrentPosition(
+        (data: GeolocationPosition) => {
+          if (data) {
+            const region: Region = {
+              latitude: data.coords.latitude,
+              longitude: data.coords.longitude,
+              latitudeDelta: 0.02,
+              longitudeDelta: 0.02,
+            }
+            callback(region)
+          }
+        },
+        () => {
+          callback(undefined)
+        },
+        { timeout: 5000 },
+      )
+    } catch (e) {
+      callback(undefined)
+    }
+  }
+
   const requestLocationPermission = useCallback(() => {
-    const asyncRequestLocationPermission = async () => {
-      try {
-        const granted = await PermissionsAndroid.request(
-          PermissionsAndroid.PERMISSIONS.ACCESS_FINE_LOCATION,
-          {
-            title: LL.MapScreen.locationPermissionTitle(),
-            message: LL.MapScreen.locationPermissionMessage(),
-            buttonNeutral: LL.MapScreen.locationPermissionNeutral(),
-            buttonNegative: LL.MapScreen.locationPermissionNegative(),
-            buttonPositive: LL.MapScreen.locationPermissionPositive(),
-          },
-        )
-        if (granted === PermissionsAndroid.RESULTS.GRANTED) {
-          console.debug("You can use the location")
+    const permittedResponse = () => {
+      getUserRegion(async (region) => {
+        if (region) {
+          setUserLocation(region)
+          setIsLoadingLocation(false)
         } else {
-          console.debug("Location permission denied")
+          setLocationDenied(true)
         }
-      } catch (err: unknown) {
-        if (err instanceof Error) {
-          crashlytics().recordError(err)
-        }
-        console.debug(err)
-      }
+      })
     }
-    asyncRequestLocationPermission()
-    // disable eslint because we don't want to re-run this function when the language changes
+
+    const negativeResponse = (error: GeolocationPermissionNegativeError) => {
+      console.debug("Permission location denied: ", error)
+      setLocationDenied(true)
+    }
+
+    Geolocation.requestAuthorization(permittedResponse, negativeResponse)
```

Comment:
I've yet to try the PR locally, but what is this permission for? is it to show the current location of the user on the map? I think we should "wait for permission" in that case following the user pushing some button to give us the permission to ask for his precise geo localization. because asking for precise location for a bitcoin wallet is something bitcoiners can be wary of. 

maybe this could be done with another PR to not get this PR increase too much in scope

Reply:
This behaviour was always in the app, it was just done before with the AndroidPermissions module instead (and iOS was done automatically). I agree though we can add that button later on so it's less pushy.

Reply:
> This behaviour was always in the app.

maybe I'm not sure to understand why the package `@react-native-community/geolocation` is necessary if this was the behavior before? (it's been years since I've coded this and I don't really remember all the details)

Reply:
> > This behaviour was always in the app.
> 
> maybe I'm not sure to understand why the package `@react-native-community/geolocation` is necessary if this was the behavior before? (it's been years since I've coded this and I don't really remember all the details)

So the app in production right now gets permission to retrieve the user's location, but never actually does anything about it. Geolocation get's the user's coordinates after permission is given. And an added feature of the Geolocation library is that it asks for permissions automatically (which is why I was able to remove the AndroidPermissions section from before).

Reply:
It's a small thing, but kind of the main point of this PR.

The app in prod currently gets the user's location permission and then does nothing about it. It does however give you access to that button in the top-right corner which will go to the user's location.

Geolocation allows for us to get the user's location before the MapView is created. Therefore it's required if we want to center on the user's location for the initial rendering.